[Bug]: When there are multiple types of makefiles the order is not respected

### Bug Description

When there are multiple makefiles in a folder lazymake will not address the correct one.

### Steps to Reproduce

1) Have one GNUmakefile and one Makefile in a folder
GNUmakefile contents:
```
build:
        touch GNUmakefile.txt
```
Makefile contents:
```
build: ## touch makefile.txt
	touch makefile.txt
run:
	@echo "Makefile is working!"
```



### Expected Behavior

Run `lazymake`.
On selecting `build` the Makefile.txt gets created.
On selecting run I should run the `run` step.

This also happens when explicitly running `lazymake -f Makefile`

### Actual Behavior

Run `lazymake`.
On selecting `build` the GNUmakefile.txt gets created.
On selecting run I get the error:
`make: *** No rule to make target 'run'.  Stop.`

This also happens when explicitly running `lazymake -f Makefile`

### lazymake Version

v0.1.0

### Operating System

Fedora 42

### Makefile Sample (if relevant)

```makefile

```

### Configuration (if relevant)

```yaml

```

### Additional Context

I think adding `make -f Makefile *command*`  for every makefile could be useful

--- a/internal/executor/executor.go
+++ b/internal/executor/executor.go
@@ -15,9 +15,9 @@ type Result struct {
 	EndTime   time.Time // When execution ended
 }
 
-func Execute(target string) Result {
+func Execute(target, makefilePath string) Result {
 	start := time.Now()
-	cmd := exec.Command("make", target)
+	cmd := exec.Command("make", "-f", makefilePath, target)
 	output, err := cmd.CombinedOutput()
 	end := time.Now()
 	duration := end.Sub(start)

--- a/internal/executor/executor_test.go
+++ b/internal/executor/executor_test.go
@@ -26,7 +26,7 @@ test:
 	os.Chdir(tempDir)
 
 	// Execute the target
-	result := Execute("test")
+	result := Execute("test", makefile)
 
 	// Verify success
 	if result.Err != nil {
@@ -65,7 +65,7 @@ fail:
 	os.Chdir(tempDir)
 
 	// Execute the failing target
-	result := Execute("fail")
+	result := Execute("fail", makefile)
 
 	// Verify failure
 	if result.Err == nil {
@@ -93,7 +93,7 @@ func TestExecuteNonExistentTarget(t *testing.T) {
 	os.Chdir(tempDir)
 
 	// Execute non-existent target
-	result := Execute("nonexistent")
+	result := Execute("nonexistent", makefile)
 
 	// Verify error
 	if result.Err == nil {
@@ -130,7 +130,7 @@ echo:
 	defer os.Chdir(oldDir)
 	os.Chdir(tempDir)
 
-	result := Execute("echo")
+	result := Execute("echo", makefile)
 
 	// Verify output contains all lines
 	expectedLines := []string{"line 1", "line 2", "line 3"}
@@ -159,7 +159,7 @@ slow:
 	defer os.Chdir(oldDir)
 	os.Chdir(tempDir)
 
-	result := Execute("slow")
+	result := Execute("slow", makefile)
 
 	// Verify timing fields are set
 	if result.StartTime.IsZero() {
@@ -254,7 +254,7 @@ fail255:
 			defer os.Chdir(oldDir)
 			os.Chdir(tempDir)
 
-			result := Execute(tt.target)
+			result := Execute(tt.target, makefile)
 
 			if (result.Err != nil) != tt.wantError {
 				t.Errorf("Execute() error = %v, wantError %v", result.Err, tt.wantError)
@@ -285,7 +285,7 @@ mixed:
 	defer os.Chdir(oldDir)
 	os.Chdir(tempDir)
 
-	result := Execute("mixed")
+	result := Execute("mixed", makefile)
 
 	// CombinedOutput should contain both stdout and stderr
 	if !contains(result.Output, "stdout message") {
@@ -314,7 +314,7 @@ test:
 	defer os.Chdir(oldDir)
 	os.Chdir(tempDir)
 
-	result := Execute("test")
+	result := Execute("test", makefile)
 
 	// Verify all fields in Result struct are populated correctly
 	if result.Output == "" {
@@ -381,7 +381,7 @@ three:
 	// Test executing different targets sequentially
 	targets := []string{"one", "two", "three"}
 	for _, target := range targets {
-		result := Execute(target)
+		result := Execute(target, makefile)
 
 		if result.Err != nil {
 			t.Errorf("Execute(%q) error: %v", target, result.Err)
@@ -415,7 +415,7 @@ concurrent:
 
 	for i := 0; i < 5; i++ {
 		go func() {
-			result := Execute("concurrent")
+			result := Execute("concurrent", makefile)
 			done <- result
 		}()
 	}
@@ -439,7 +439,7 @@ func TestExecute_CommandNotFound_SetsAllFields(t *testing.T) {
 	os.Setenv("PATH", "/nonexistent") // Should fail to find 'make'
 
 	start := time.Now()
-	result := Execute("anytarget")
+	result := Execute("anytarget", "makefile")
 	end := time.Now()
 
 	if result.Err == nil {
@@ -502,7 +502,7 @@ func TestExecute_ResultFields_AlwaysSetEvenOnFailure(t *testing.T) {
 			defer os.Chdir(oldDir)
 			os.Chdir(tempDir)
 
-			res := Execute(tc.target)
+			res := Execute(tc.target, f)
 
 			if tc.shouldErr && res.Err == nil {
 				t.Error("expected error, got nil")
@@ -559,6 +559,6 @@ bench:
 
 	b.ResetTimer()
 	for i := 0; i < b.N; i++ {
-		Execute("bench")
+		Execute("bench", makefile)
 	}
 }

--- a/internal/tui/update.go
+++ b/internal/tui/update.go
@@ -248,7 +248,7 @@ func (m Model) handleTargetSelection() (tea.Model, tea.Cmd) {
 	m.ExecutionElapsed = 0
 
 	return m, tea.Batch(
-		executeTarget(target.Name),
+		executeTarget(target.Name, m.MakefilePath),
 		tickTimer(),
 		m.Spinner.Tick,
 	)
@@ -684,7 +684,7 @@ func (m Model) updateConfirmDangerous(msg tea.Msg) (tea.Model, tea.Cmd) {
 				m.ExecutionStartTime = time.Now()
 				m.ExecutionElapsed = 0
 				return m, tea.Batch(
-					executeTarget(target.Name),
+					executeTarget(target.Name, m.MakefilePath),
 					tickTimer(),   // Start timer
 					m.Spinner.Tick, // Start spinner animation
 				)
@@ -707,9 +707,9 @@ type executeFinishedMsg struct {
 // Custom message for timer ticks
 type timerTickMsg struct{}
 
-func executeTarget(target string) tea.Cmd {
+func executeTarget(target, makefilePath string) tea.Cmd {
 	return func() tea.Msg {
-		result := executor.Execute(target)
+		result := executor.Execute(target, makefilePath)
 		return executeFinishedMsg{result: result}
 	}
 }

